config/check_dependencies.py

```diff
@@ -0,0 +1,138 @@
+#
+#  SPDX-License-Identifier: BSD-3-Clause
+#
+#  SPDX-License-Identifier: BSD-3-Clause
+
+import getopt
+import json
+import sys
+from os.path import exists
+
+filename = "check_dependies.py"
+
+
+class ComponentValidator:
+    def __init__(self, allow_component, sbom_component):
+        self.allow = allow_component
+        self.sbom = sbom_component
+        self.errors = []
+
+    def validate(self, product=False):
+        self.__name()
+        self.__licenses()
+        if not product:
+            self.__version()
+        return self.errors
+
+    def __name(self):
+        if self.allow["name"] != self.sbom["name"]:
+            self.append_error("name", self.allow["name"], self.sbom["name"])
+
+    def __version(self):
+        if self.allow["version"] != self.sbom["version"]:
+            self.append_error("version", self.allow["version"], self.sbom["version"])
+
+    def __licenses(self):
+        allow_licenses = self.allow["licenses"]
+        sbom_licenses = self.sbom.get("licenses") if self.sbom.get("licenses") else []
+
+        for sbom_lic in sbom_licenses:
+            sbom_id = sbom_lic["license"].get("id")
+            if not sbom_id:
+                sbom_id = sbom_lic["license"].get("name")
+
+            id_list = sbom_id in [lic["license"].get("id") for lic in allow_licenses]
+            if not id_list:
+                self.append_error("license", "to be found in {}".format(allow_licenses), sbom_id)
+
+    def append_error(self, key, expect, current):
+        self.errors.append("Component '{}':'{}' expects '{}' but was '{}'."
+                           .format(self.allow["group"], key, expect, current))
+
+
+def read_json(file_path):
+    if not exists(file_path):
+        print("{} does not exist.".format(file_path))
+        sys.exit(2)
+
+    with open(file_path) as f:
+        return json.load(f)
+
+
+def validate(allow_list, sbom):
+    allow_json = read_json(allow_list)
+    sbom_json = read_json(sbom)
+    error_collection = []
+
+    # product
+    validator = ComponentValidator(allow_json["component"], sbom_json["metadata"]["component"])
+    product_err = validator.validate(True)
+    if product_err:
+        error_collection.append(product_err)
+
+    # dependencies
+    if len(allow_json["components"]) != len(sbom_json["components"]):
+        allowed_groups = [item["name"] for item in allow_json["components"]]
+        sbom_groups = [item["name"] for item in sbom_json["components"]]
+        error_collection.append("Number of licenses expects {} but was {}.".format(len(allow_json["components"]),
+                                                                                   len(sbom_json["components"])))
+        missing_allowed = set(sbom_groups).difference(allowed_groups)
+        missing_sbom = set(allowed_groups).difference(sbom_groups)
+        if missing_sbom:
+            error_collection.append("License {} not found in sbom but in allow list.".format(missing_sbom))
```

Comment:
If this is an error, is the allow list really an allow list or rather a "checklist"?